src/main/scala/mesosphere/marathon/core/task/tracker/impl/steps/NotifyRateLimiterStepImpl.scala

```diff
@@ -18,35 +18,34 @@ class NotifyRateLimiterStepImpl @Inject() (
   override def name: String = "notifyRateLimiter"
 
   override def processUpdate(
-    timestamp: Timestamp, appId: PathId, maybeTask: Option[MarathonTask], status: TaskStatus): Future[_] = {
+    timestamp: Timestamp, appId: PathId, task: MarathonTask, status: TaskStatus): Future[_] = {
     import org.apache.mesos.Protos.TaskState._
 
     status.getState match {
       case TASK_ERROR | TASK_FAILED | TASK_FINISHED | TASK_KILLED | TASK_LOST =>
-        notifyRateLimiter(appId, status, maybeTask)
+        notifyRateLimiter(appId, status, task)
 
       case _ =>
         Future.successful(())
     }
-
   }
 
   private[this] def notifyRateLimiter(
     appId: PathId,
     status: TaskStatus,
-    maybeTask: Option[MarathonTask]): Future[_] = {
-    maybeTask match {
-      case Some(task) if status.getState != TaskState.TASK_KILLED =>
-        import scala.concurrent.ExecutionContext.Implicits.global
-        appRepository.app(appId, Timestamp(task.getVersion)).map { maybeApp =>
-          // It would be nice if we could make sure that the delay gets send
-          // to the AppTaskLauncherActor before we continue but that would require quite some work.
-          //
-          // In production, it should not matter if we do not delay some task launches so much.
-          maybeApp.foreach(launchQueue.addDelay)
-        }
-      case _ =>
-        Future.successful(())
+    task: MarathonTask): Future[_] = {
+    if (status.getState != TaskState.TASK_KILLED) {
+      import scala.concurrent.ExecutionContext.Implicits.global
+      appRepository.app(appId, Timestamp(task.getVersion)).map { maybeApp =>
+        // It would be nice if we could make sure that the delay gets send
+        // to the AppTaskLauncherActor before we continue but that would require quite some work.
+        //
+        // In production, it should not matter if we do not delay some task launches so much.
```

Comment:
The delay is a result of an effect that happened in the past. The message abut that effect also takes time. I am tempted to say: please remove this comment, because we react as prompt as possible...